src/relayer/Relayer.ts

```diff
@@ -306,6 +308,105 @@ export class Relayer {
     this.clients.tokenClient.decrementLocalBalance(deposit.destinationChainId, deposit.destinationToken, fillAmount);
   }
 
+  // Strategy for requesting refunds: Query all refunds requests, and match them to fills.
+  // The remaining fills are eligible for new requests.
+  async requestRefunds(): Promise<void> {
+    const { multiCallerClient, ubaClient } = this.clients;
+    const spokePoolClients = Object.values(this.clients.spokePoolClients);
+
+    await ubaClient.update();
+
+    const eligibleFillsByRefundChain: { [chainId: number]: FillWithBlock[] } = Object.fromEntries(
+      spokePoolClients.map(({ chainId }) => [chainId, []])
+    );
+
+    const refundRequestsByDstChain: { [chainId: number]: RefundRequestWithBlock[] } = {};
+
+    spokePoolClients.forEach((spokePoolClient) => {
+      // For each refund/repayment Spoke Pool, group its set of by corresponding destination chain.
+      spokePoolClient.getRefundRequests().forEach((refundRequest: RefundRequestWithBlock) => {
+        if (
+          refundRequest.relayer === this.relayerAddress &&
+          ubaClient.refundRequestIsValid(spokePoolClient.chainId, refundRequest)
+        ) {
+          refundRequestsByDstChain[refundRequest.destinationChainId].push(refundRequest);
+        }
+      });
+    });
+
+    for (const spokePoolClient of spokePoolClients) {
+      const { chainId: destinationChainId } = spokePoolClient;
+      const fills = this.findEligibleFills(spokePoolClient, refundRequestsByDstChain[destinationChainId]);
+      fills.forEach((fill) => eligibleFillsByRefundChain[fill.repaymentChainId].push(fill));
+    }
+
+    // Enqueue a refund for each eligible fill.
+    Object.values(eligibleFillsByRefundChain).forEach((fills) => {
+      fills.forEach((fill) => {
+        this.logger.debug({ at: "Relayer::requestRefunds", message: "Requesting refund for fill.", fill });
+        this.requestRefund(fill);
+      });
+    });
+
+    await multiCallerClient.executeTransactionQueue();
+  }
+
+  findEligibleFills(
```

Comment:
```suggestion
  findFillsRequiringSeparateRefundRequest(
```
I think this name is more clear, but not perfect. I think "eligible" is just kind of vague